tests/progs/test_jubjub.py

```diff
@@ -216,3 +212,28 @@ def test_basic_point_functionality():
             await p1.montgomery_mul(5))
 
     await run_test_prog(_prog, test_preprocessing)
+
+
+@mark.asyncio
+async def test_share_mul(test_preprocessing):
+
+    p = TEST_POINTS[1]
+    multiplier = 7
+    multiplier_bin = bin(multiplier)[2:]
+
+    async def _prog(context):
+        print("type(SharedIdeal(TEST_CURVE)): ", type(SharedIdeal(TEST_CURVE)))
+
+        p_ = await SharedPoint.from_point(context, p)
+        multiplier_ = [context.Share(int(i)) for i in multiplier_bin]
+
+        p1_, p2_ = await gather(
+            share_mul(context, p, multiplier_),
+            p_.mul(multiplier))
+        assert await shared_point_equals(p1_, p2_)
+
+        q1_ = await share_mul(context, Ideal(TEST_CURVE), multiplier_)
+        q2_ = SharedIdeal(TEST_CURVE)
```

Comment:
oh wait, q2_ should just be an `Ideal`-- `SharedIdeal` is meant for use with `SharedPoint`, which is not used with what you're implementing. I should document this.